### PR TITLE
Updated build workflow for releases

### PR DESCRIPTION
**Pull request by A-Biggs**
This workflow creates a 2 zip file (one for the OVAL schemas, and another for the OVAL HTML docs) and attaches them to the Release in GitHub.

See the [Adam 5.13.0 Build Pipeline Test](https://github.com/A-Biggs/OVAL/releases/tag/Adam5.13.0Test) at https://github.com/A-Biggs/OVAL/releases for an example of what is generated by this workflow for a release.

`<a href="http://oval.cisecurity.org/terms">OVAL Terms of Use</a>` should be updated in the rsrc/index.html file with a new link as that one is currently broken.

### CODE REVIEW

**Comment by solind** [comment]
Hi @A-Biggs, the [oval_docs-5.13.0.zip](https://github.com/A-Biggs/OVAL/releases/download/Adam5.13.0Test/oval_docs-5.13.0.zip) generated by your repo doesn't appear to contain the `index.html` file.  Any idea why it doesn't?

File: .github/workflows/buildRelease.yml
Comment (DeafBDoor) [inline .github/workflows/buildRelease.yml:33]: This requires an existing release. I'd suggest you use a release create instead. Even if you don't want to automatically release, you can set the command to create a draft release:

`gh release create $TAG "$DOC_DIR".zip -d --generate-notes --verify-tag --repo $OWNER/$REPO`

Comment (DeafBDoor) [reply .github/workflows/buildRelease.yml:33]: There is also the option to verify if a release already exists, then just update the assets.
Something like:
```
if $(gh release list | head -n2 | cut -d$'\t' -f3 | grep -w -q "$TAG"); then 
    gh release upload $TAG schemas-${VERSION}.zip oval_docs-${VERSION}.zip --clobber --repo $OWNER/$REPO
else
    gh release create $TAG schemas-${VERSION}.zip oval_docs-${VERSION}.zip -d --generate-notes --verify-tag --repo $OWNER/$REPO
fi
```

File: .github/workflows/buildRelease.yml
Comment (DeafBDoor) [inline .github/workflows/buildRelease.yml:31]:
```suggestion
```

**Review by DeafBDoor — APPROVED** [review]

**Comment by vanderpol** [comment]
@A-Biggs I'd like to confirm @solind question about the lack of an index.html, it definitely would be useful.   Otherwise it looks good.

**Comment by A-Biggs** [comment]
> Hi @A-Biggs, the [oval_docs-5.13.0.zip](https://github.com/A-Biggs/OVAL/releases/download/Adam5.13.0Test/oval_docs-5.13.0.zip) generated by your repo doesn't appear to contain the `index.html` file. Any idea why it doesn't?

Good catch.  I missed including `index.html` in the zip command when I changed it to using specific files names instead of `docs/*.html`.  I'll fix this.

Any ideas what this broken link (`<a href="http://oval.cisecurity.org/terms">OVAL Terms of Use</a>`)should be updated to in the index.html file?

File: .github/workflows/buildRelease.yml
Comment (A-Biggs) [reply .github/workflows/buildRelease.yml:33]: @DeafBDoor right now the triggers for this workflow is such that the release already exists, so unless a manual trigger is added, the `gh release create` stuff isn't needed.  Not sure if there is really a need for a manual trigger for this workflow, drafting a release with the `/releases/new` URL and unchecking the "Set as a pre-release" and "Set as the latest release" boxes, and clicking publish.  Saving the release as a draft doesn't trigger the workflow, but I don't think it should, as I see that state as where one edits the description and release notes for a release (the auto-generated and attached source files also don't get attached in that state).

If people think there is a need/use case to have a manual trigger for this workflow, I can add it.

**Comment by vanderpol** [comment]
> > Hi @A-Biggs, the [oval_docs-5.13.0.zip](https://github.com/A-Biggs/OVAL/releases/download/Adam5.13.0Test/oval_docs-5.13.0.zip) generated by your repo doesn't appear to contain the `index.html` file. Any idea why it doesn't?
> 
> Good catch. I missed including `index.html` in the zip command when I changed it to using specific files names instead of `docs/*.html`. I'll fix this.
> 
> Any ideas what this broken link (`<a href="http://oval.cisecurity.org/terms">OVAL Terms of Use</a>`)should be updated to in the index.html file?

Can you point it to:  https://github.com/OVAL-Community/OVAL/blob/master/terms-of-use.rst?

If it needs to be HTML, I can make an HTML version as well.

**Comment by A-Biggs** [comment]
@vanderpol I just updated the Terms of Use in `index.html` with the link you provided.

**Comment by vanderpol** [comment]
@A-Biggs lets see another test build after your latest updates on your fork, and I'll merge this into OVAL master.

File: rsrc/saxon9he.jar
Comment (DeafBDoor) [inline rsrc/saxon9he.jar:1]: @A-Biggs instead of adding the saxon9he.jar to the git repository, I think it would be better if you download it from a .jar repository. For instance, the MAVEN one:

```
mkdir -p rsrc
curl https://repo1.maven.org/maven2/net/sf/saxon/Saxon-HE/9.9.1-8/Saxon-HE-9.9.1-8.jar --output rsrc/saxon9he.jar
```

I tested the aforementioned file and it worked great. That also would make it easier to update the jar dependency when needed.

**Comment by vanderpol** [comment]
Just checking to see if this is ready for merging, or if any additional updates are still pending?

**Comment by vanderpol** [comment]
Thanks the the clarification @DeafBDoor, as I'm looking for a sanity review on the merge of develop to master, we have some time, I'll wait for @A-Biggs confirmation, then merge this PR

**Comment by A-Biggs** [comment]
@vanderpol I just pushed a changed based on @DeafBDoor's suggestion.  Here is the result of the latest run of the workflow, https://github.com/A-Biggs/OVAL/releases/tag/Adam5.13.0Test2.

**Comment by vanderpol** [comment]
@A-Biggs, looks great.  I'll merge anytime you all say so.  I have a question on the index.html, I'm wondering what it will do with the new AWS and PANOS schemas?  Can we try this process out on a branch of develop to see if we need some hard coding of that to know where it falls on the index.html?  I'm curious how the index.html knows which schema's are Network Device Schemas etc..

Also, something to consider... how much work would it be to have each html page have an outline section with hyperlinks to each "Test", "Item" or "Type"?   Not something we have to get into 5.12, but if it's a trivial change, it might be useful.

**Comment by A-Biggs** [comment]
@vanderpol IMO, this PR is good to merge.

index.html is just a static file (minus the version number being replaced), so when new schemas are added, the index.html file will need to be edited to include those schemas in the appropriate spots.  There were items that were included in the index.html that was in the Joval repo that weren't a part of the official schema, for which I removed, one being [PanOS](https://github.com/joval/jOVAL/blob/1c5f9da747a1bca8903fc6e28bce2182dacc44a9/scap/rsrc/index.html#L198-L202).


> how much work would it be to have each html page have an outline section with hyperlinks to each "Test", "Item" or "Type"? Not something we have to get into 5.12, but if it's a trivial change, it might be useful.

My knowledge of HTML isn't that extensive, so that might be better answered by someone else, but my gut is saying it's a bit more than trivial, but it shouldn't be too hard for someone that knows XSL and HTML.  If no one else jumps on this, I might be able to take a look at this at some point.

**Comment by vanderpol** [comment]
Thanks @A-Biggs  and @DeafBDoor, merged it in and we can update the static index.html file with aws etc.. after we merge develop into master, hopefully by tomorrow.

**Comment by solind** [comment]
> > how much work would it be to have each html page have an outline section with hyperlinks to each "Test", "Item" or "Type"? Not something we have to get into 5.12, but if it's a trivial change, it might be useful.
> 
> My knowledge of HTML isn't that extensive, so that might be better answered by someone else, but my gut is saying it's a bit more than trivial, but it shouldn't be too hard for someone that knows XSL and HTML. If no one else jumps on this, I might be able to take a look at this at some point.

You could conceivably write a script to generate the index.html from the project files.  To do so, you'd have to move the data (that's in index.html's HTML table of contents) from the static index.html into all those project files somewhere, then have the script collect it all and generate the table.  I don't really think it'd be worth the effort.

**Pull request merged**
